src/http-proxy.js

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import URL from 'url';
+
+import UploadError from './upload-error';
+import ErrorCodes from './error-codes';
+import { getBasicAuth } from './http-utils';
+
+const PRIVATE = Symbol('PRIVATE');
+
+/**
+ * Represents information about the proxy to use when sending HTTP requests.
+ */
+export default class HttpProxy {
+    /**
+     * Constructs new proxy information using the given proxy URL. By default, authentication will not be used with the proxy.
+     * @param {string} proxyUrl Full URL of the HTTP proxy to be used. Example: http://localhost:5000.
+     */
+    constructor(proxyUrl) {
+        this[PRIVATE] = {
+            proxyUrl
+        };
+    }
+
+    /**
+     * Retrieves the URL of the proxy as it was provided on construction.
+     * @returns {URL} Full URL of an HTTP proxy.
+     */
+    getUrl() {
+        const { proxyUrl } = this[PRIVATE];
```

Comment:
Why not use `this[PRIVATE].proxyURL` and have one variable less in this function?

Reply:
Just personal preference. I prefer setting default values with destructuring (i.e. `const { proxyUrl = 'default value' } = this[PRIVATE];`). True, there's no default value here, but the habit is still there. IMO it doesn't change the behavior of the code, so it just comes down to preference.